src/controllers/forecast.mjs

```diff
@@ -1,23 +1,38 @@
 import { getForecastByCity } from "../services/forecast.mjs";
-import { getFileData } from "../utils/fileHelper.mjs";
-import { CITIES_FILE_PATH } from "../../config.mjs";
+import { getFileData, writeFile } from "../utils/fileHelper.mjs";
+import { CITIES_FILE_PATH, WEATHER_LOG_FILE_PATH } from "../../config.mjs";
+
+const checkUpdate = async (city) => {
+  const cityName = city.name.toLowerCase();
+  const weatherLog = await getFileData(WEATHER_LOG_FILE_PATH);
+  const cityWeather = weatherLog[cityName];
+  const twoHoursAgo = Date.now() - 2 * 60 * 60 * 1000;
+  const needUpdate =
+    !cityWeather || new Date(cityWeather.lastUpdate).getTime() < twoHoursAgo;
+
+  if (needUpdate) {
+    const forecast = await getForecastByCity(city);
+    weatherLog[cityName] = {
+      ...forecast,
+    };
+  }
+  await writeFile(WEATHER_LOG_FILE_PATH, weatherLog);
+};
```

Comment:
I would propose to manage this question on service level
Service responsible to return us the data, no matter how

So, in controller we have only business logic in terms of "I have this request, I know that I need to work with city, I know that I can ask service for a forecast and I don't care how exactly. After that I need to return data that service provides to me"

We can also introduce one more util that would be responsible to check expiration
It should know about time-shift condition (grabbing info about outdate amount of ms from config `const EXPIRATION_IN_MS = 2 * 60 * 60 * 1000`)
We might have a function there (`isExpirationValid`) that doesn't care about data itself, it cares only about input number property and returns true/false